addons/web/static/tests/core/dropdown_tests.js

```diff
@@ -1313,4 +1315,63 @@ QUnit.module("Components", ({ beforeEach }) => {
             '<div class="o-dropdown dropdown o-dropdown--no-caret"><button class="dropdown-toggle" disabled="" tabindex="0" aria-expanded="false"></button></div>'
         );
     });
+
+    QUnit.test("don't close dropdown outside the active element", async (assert) => {
+        // This test checks that if a dropdown element opens a dialog with a dropdown inside,
+        // opening this dropdown will not close the first dropdown.
+        patchWithCleanup(Dropdown.prototype, {
+            close() {
+                assert.notOk(true, "dropdown should not be closed");
+                this._super();
+            },
+        });
+
+        class CustomDialog extends Component {}
+        CustomDialog.template = xml`
+            <Dialog title="'Welcome'">
+                <Dropdown>
+                    <DropdownItem>Item</DropdownItem>
+                </Dropdown>
+            </Dialog>`;
+        CustomDialog.components = { Dialog, Dropdown, DropdownItem };
+
+        const mainComponentRegistry = registry.category("main_components");
+        clearRegistryWithCleanup(mainComponentRegistry);
+        serviceRegistry.add("dialog", dialogService);
+        serviceRegistry.add("l10n", makeFakeLocalizationService());
+
+        class PseudoWebClient extends Component {
+            setup() {
+                this.Components = mainComponentRegistry.getEntries();
+            }
+            clicked() {
+                env.services.dialog.add(CustomDialog);
+            }
+        }
+        PseudoWebClient.template = xml`
+                <div>
+                    <div>
+                        <t t-foreach="Components" t-as="C" t-key="C[0]">
+                            <t t-component="C[1].Component" t-props="C[1].props"/>
+                        </t>
+                    </div>
+                    <div>
+                        <Dropdown>
+                            <button class="click-me" t-on-click="clicked">Click me</button>
+                        </Dropdown>
+                    </div>
+                </div>
+            `;
+        PseudoWebClient.components = { Dropdown };
+
+        env = await makeTestEnv();
+        await mount(PseudoWebClient, target, { env });
+        await click(target, "button.dropdown-toggle");
+        assert.containsOnce(target, ".dropdown-menu");
+        await click(target, "button.click-me");
+        await nextTick();
```

Comment:
should not be necessary (?) What am I missing?

Reply:
Indeed